### ISSUE TITLE
Port HA Kubernetes from Kubernetes-Ansible Project

### ISSUE THREAD

**Issue by PeterLamar** [CiscoCloud/microservices-infrastructure#891]
@altvnk just did some good work getting HA and upgrading the automation there. Perhaps we can examine it and port the effort to here? @Zogg @stevendborrelli

**Comment by stevendborrelli** [comment]
I agree with this.

**Comment by Zogg** [comment]
HA is indeed a welcome guest.

**Comment by danehans** [comment]
Instead of porting the work, I suggest migrating Mantl to use the upstream project [1].

[1] https://github.com/kubernetes/contrib/tree/master/ansible

If functionality is missing from [1], push the changes upstream and only maintain code/playbook functionality that is Cisco-specific or contains Cisco IP.

**Comment by Zogg** [comment]
Currently, I'v ported https://github.com/CiscoCloud/mantl-kubernetes into Mantl.

The suggestion sounds good, but the reality is such that we want to support many orchestration backends: Mesos, K8s, Nomad, also, we have different software stacks (calico instead of flannel for starters), which leaves only the core k8s ansible tasks intact, even manifests of various services are modified here and there. So at least for now, it is easier to port the work, instead of migrate.

**Comment by danehans** [comment]
@Zogg I agree that porting the upstream work is easier. I also understand differences exist between Mantl's fork and the upstream project. My recommendation is for the long-term health of the project.

**Comment by siddharthist** [comment]
@danehans It doesn't appear that the upstream code is well supported (lots of TODOs, a task named "HACK"). Additionally, the roles aren't uploaded to Ansible Galaxy. Furthermore, their roles generate their own certs, while we want to unify all of our components to use the ones we generate (for ease of rotation). Until all those issues are sorted out, I think we'll just have to use our own code.
